extensions/lifecycle/conductor/LifecycleConductor.js

```diff
@@ -325,12 +367,15 @@ class LifecycleConductor {
                                 }
 
                                 isTruncated = result.IsTruncated;
-                                nEnqueued += result.Contents.length;
 
                                 result.Contents.forEach(o => {
                                     marker = o.key;
                                     const [canonicalId, bucketName] = marker.split(constants.splitter);
-                                    queue.push({ canonicalId, bucketName });
+
+                                    if (!this._isBlacklisted(canonicalId, bucketName)) {
+                                        nEnqueued += 1;
+                                        queue.push({ canonicalId, bucketName });
+                                    }
```

Comment:
Potential optimization here: at the end of the loop, if the last entry was a denied account id, there's a good chance that the next "page" also starts with the same account, and with bad luck, could go on for several thousands of pages.

We could cheat in this specific case by overriding the marker, to start listing at the next canid (would be trivial with an allowlist-style rule but here it needs a trick). For example:
- last entry of `Contents` is `abcdef:bucketfoo`
- (next list probably starts with `abcdef:bucketbar`)
- (could be several millions of entries to list and evaluate)
- after the `Contents.forEach` loop, if the last entry was skipped:
- overwrite the marker with `abcdef;` (because `;` is immediately after `:`)
- the next listing will start at the next available unskipped account